DevelopADC and PWM providers added

MahmoudGSaleh opened the pull request.

MahmoudGSaleh committed:
Merge remote-tracking branch 'refs/remotes/ms-iot/develop' into develop

MahmoudGSaleh committed:
Merge remote-tracking branch 'refs/remotes/ms-iot/develop' into develop

MahmoudGSaleh committed:
Update GPIO sample

Indicate if Lightning provider or the default one is enabled

MahmoudGSaleh committed:
Merge remote-tracking branch 'refs/remotes/ms-iot/develop' into develop

MahmoudGSaleh committed:
Remove dependency on the low level devices contracts winmd

Additionally, this fixes an issue where the native .net native tool
chain could not be used. So, re-enabling the usage of the .net native
compiler for release builds.

MahmoudGSaleh committed:
Merge branch 'develop' of https://github.com/msalehmsft/BusProviders into develop

MahmoudGSaleh committed:
Add PWM provider support

This update to the provider library, adds Lightning PWM provider for
PCA9685 as well as an emulated PWM software provider that uses Lightning
GPIO.
Additionally, there are several fixes to samples and library code.

MahmoudGSaleh committed:
Add ADC provider

MahmoudGSaleh committed:
Merge remote-tracking branch 'refs/remotes/ms-iot/develop' into develop

bfjelds commented on Microsoft.IoT.Lightning.Providers/Providers/Microsoft.Iot.Lightning.Providers.vcxproj.filters:
```diff
@@ -24,11 +24,10 @@
     <ClInclude Include="Dummy.h">
```
is dummy still needed after 10586 is released?

MahmoudGSaleh replied:
Yes, it needed, as the toolchain fix won't be part of update 1.

bfjelds commented on Microsoft.IoT.Lightning.Providers/AdcConsumer/AdcConsumer.csproj:
```diff
@@ -0,0 +1,140 @@
+    <AssemblyName>AdcConsumer</AssemblyName>
+    <DefaultLanguage>en-US</DefaultLanguage>
+    <TargetPlatformIdentifier>UAP</TargetPlatformIdentifier>
+    <TargetPlatformVersion>10.0.10240.0</TargetPlatformVersion>
```
10586?

MahmoudGSaleh replied:
Fixed

bfjelds commented on Microsoft.IoT.Lightning.Providers/AdcConsumer/StartupTask.cs:
```diff
@@ -0,0 +1,50 @@
+                return;
+            }
+
+            // Use the PAC9685 PWM provider
```
comment refers to PWM rather than adc?

MahmoudGSaleh replied:
Fixed.

MahmoudGSaleh committed:
Fix samples

MahmoudGSaleh committed:
Update TargetPlatformMinVersion to 10586.0

Target TH2 version.
Additionally, update the nuget version numbers to latest available.

MahmoudGSaleh committed:
NuGet package added for Lightning Providers WinRT component
